### DISCUSSION TITLE
Behavior of the library when localeCookie is set to false

### DISCUSSION THREAD

**Discussion by seloElo** [amannn/next-intl#1491] [Q&A] [answered]
Hello,

Was playing with the new update today, and I tried setting the localeCookie option to false to see what was the impact on my apps. 

I noticed that the accept-language header in any request will always be the defaultLocale with no cookie, even if a different language was correctly set on the client using url for example. Is that the expected behavior ?

If not, happy to open a bug ticket, but I was wondering if it was a limitation to this option.

**Comment by amannn** [comment]
Hmm, nope—that doesn't sound right to me. The `localeCookie` option should not have an effect on the negotiation based on the `accept-language` header.

Can you share your routing config and how you test the behavior?

We have quite an extensive test suite for the middleware (see [`middleware.test.tsx`](https://github.com/amannn/next-intl/blob/main/packages/next-intl/src/middleware/middleware.test.tsx)), you can also experiment with a unit test there if you have a rather special case.

**Reply by seloElo** [reply]
TL:DR: I'm dumb.

I tried the setup with the bug report repo: https://github.com/seloElo/next-intl-bug-repro-app-router-locale

With the following routing config: 

```
export const routing = defineRouting({
  locales: ["en", "fr"],
  defaultLocale: "en",
  localeCookie: false,
});
```

![image](https://github.com/user-attachments/assets/9202a839-4d3f-4fc9-9beb-6c803292c84c)

![image](https://github.com/user-attachments/assets/0afe9311-cb08-47ea-a7de-d5b2e9f97ef3)

I'm a real newbie, I thought the middleware role was also to set the accept language for outgoing request done on the client using the fetch API. but it looks like the only way why it was working without the localeCookie was because the NEXT_LOCALE cookie was passed to the API, which redirected to the correct path. 

Why it confused me was because my app used the cookie under the hood, and my APIs where all redirected correctly.
Now I'll have to pass the correct locale in the pathname when calling the route handlers, that's it

**Reply by amannn** [reply, marked as answer]
Ah, got it! So you were calling a Route Handler from the client and relied on reading `NEXT_LOCALE` there, is that correct? Maybe you can consider an explicit contract there, e.g. by passing a `?locale=…` search param (see also: [Route Handlers](https://next-intl-docs.vercel.app/docs/environments/actions-metadata-route-handlers#route-handlers))

**Reply by seloElo** [reply]
Exactly, I was calling a route handler and was omitting to pass any locale in the url of the route handler thinking eveything was handled automatically, and I was using the param locale in the route handler definition.
Thanks to the NEXT_LOCALE I was redirected correctly but because it's gone now, I need to pass the locale in the fetch manually

**Reply by amannn** [reply]
That sounds like a reasonable approach, yes!
